shunit2_args_test.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/sh
+#
+# shunit2 unit test for running subset(s) of tests based upon command line arguments.
+# Shows how non-default tests or a arbitrary subset of tests can be run.
+#
+# Disable source following.
+#   shellcheck disable=SC1090,SC1091
+
+# Load test helpers.
+. ./shunit2_test_helpers
+
+# This test does not nomrally run because it does not begin "test*"
+# Will be run by settting the arguments to the script to include the name of this test.
+non_default_test() {
+  # arbitrary assert
+  assertTrue 0
+  # true intent is to set this variable, which will be tested below
+  NON_DEFAULT_TEST_RAN="yup, we ran"
+}
+
+# Test that the "non_default_test" ran, otherwise fail
+test_non_default_ran() {
+  assertNotNull "'non_default_test' did not run" "$NON_DEFAULT_TEST_RAN"
+}
+
+# fail if this test runs, which is shouldn't if args are set correctly.
+test_will_fail() {
+  fail "test_will_fail should not be run if arg-parsing works"
+}
+
+oneTimeSetUp() {
+  th_oneTimeSetUp
+  # prime with "null" value
+  NON_DEFAULT_TEST_RAN=""
+}
+
+# Load and run shunit2.
+# shellcheck disable=SC2034
+[ -n "${ZSH_VERSION:-}" ] && SHUNIT_PARENT=$0
+# if zero/one arguments provided, artificially set the arguments to this script, in a POSIX way
```

Comment:
As you are trying to demonstrate the new functionality of passing arguments, I don't think it makes sense to make that functionality conditional (line 42). Replace this comment with a statement of what you are doing in line 43, and remove the if/fi in 42/44.

Reply:
Actually, I intended that conditional to allow one to test the test script itself by using command-line arguments to specify individual tests.  In that case, I wouldn't want to override the user-specified tests.  Therefore, I improved the comments (see commit 298c99ef882ff09da6c99349c6442cd6ce63bb71) to better explain what I did rather than change the functionality.  Thoughts?